website/content/en/docs/setup/going-to-prod/hardening.md

```diff
@@ -0,0 +1,78 @@
+| Privilege escalation attacks | 🛡️ Do not run Vector as root<br />🛡️ Restrict the Vector service account<br />🛡️ Ensure Vector is the single tenant<br />🛡️ Disable SSH or remote access |
+| Upstream attacks | 🛡️ Review Vector’s security policy<br />🛡️ Upgrade Vector frequency |
+
+## Defense-In-Depth
+
+Vector takes a [defense-in-depth approach](https://en.wikipedia.org/wiki/Information_security#Defense_in_depth) to hardening and follows the data, process, host, and network [onion model](https://en.wikipedia.org/wiki/Onion_model).
+
+![Onion model](/img/going-to-prod/onion-model.png)
+
+### Securing the Data
+
+#### Securing Data at Rest
+
+Vector will not serialize events on disk unless you’ve configured Vector to use disk buffers or have enabled swap. Therefore, to secure data at rest, we recommend:
+
+- **Enable whole disk encryption.** Whole disk encryption moves the responsibility of disk encryption to the operating system or file system. This offers better holistic security, and is faster and more CPU efficient. Please follow your platform’s guidance for encryption at rest (i.e., [AWS](https://docs.aws.amazon.com/whitepapers/latest/efs-encrypted-file-systems/encryption-of-data-at-rest.html), [Azure](https://docs.microsoft.com/en-us/azure/security/fundamentals/encryption-atrest), [Google Cloud](https://cloud.google.com/security/encryption/default-encryption)).
+- **Disable swap.** Disabling swap prevents the operating system from overflowing Vector’s memory to disk, reducing disk exposure of your data. This is also better for performance. Vector should always have enough memory to operate without swap.
+- **Restrict Vector’s data directory.** The unprivileged Vector service account should be the only account that can read or write into Vector’s data directory (i.e., `/var/lib/vector` ).
+- **Encrypt external storage.** Finally, external storage, such as archives, should be encrypted. This can be done in a variety of ways depending on your security model but is typically achieved through server-side encryption. Please follow your storage’s guidance for encryption (i.e., [AWS S3](https://docs.aws.amazon.com/AmazonS3/latest/userguide/bucket-encryption.html), [Azure Blob Storage](https://docs.microsoft.com/en-us/azure/storage/common/storage-service-encryption), [Google Cloud Storage](https://cloud.google.com/storage/docs/encryption)).
+
+#### Securing Data in Transit
+
+- **Redact sensitive attributes.** Event attributes that hold sensitive data, such as PII, can be redacted (i.e., the VRL `redact` function).
+- **Disable core dumps.** A user who can force a core dump could access Vector’s in-flight data. Preventing core dumps is specific to your platform. On Linux setting the resource limit `RLIMIT_CORE` to `0` disables core dumps. In the systemd service unit file, setting `LimitCORE=0` will enforce this setting for the Vector service (this is done automatically when installing Vector through `apt` or `yum`).
```

Comment:
I don't think we actually set `LimitCORE` in the SystemD unit files we distribute.